app/react-native/src/server/config/defaults/webpack.config.js

```diff
@@ -3,16 +3,26 @@ import { includePaths } from '../utils';
 
 // Add a default custom config which is similar to what React Create App does.
 module.exports = storybookBaseConfig => {
-  const newConfig = storybookBaseConfig;
+  const newConfig = { ...storybookBaseConfig };
```

Comment:
actually, can you try just exporting the `app/react` version of this file directly? It looks like the configs could probably be shared directly.

this is outside the scope of your PR but I see a lot of shared server files for `react-native` and `react` as well. we could combine them as much as possible to prevent fixing problems in one while forgetting about the other @ndelangen